File: 2023/1/part_one.rb
#!/usr/bin/env ruby

def find_numbers(input)
	if
		input.match(/(\d).*(\d)/)
		$1 + $2
	else
		input.match(/(\d)/)
		$1 + $1
	end
end

sum = 0
for line in STDIN
	n = find_numbers(line)
	puts n
	sum += n.to_i
end
puts "---"
puts sum

File: 2023/14/part_one.php
#!/usr/bin/env php
<?php
declare(strict_types=1);

$sum = 0;
$handle = fopen("input.txt", "r");
$columns = array_fill(0, strlen(fgets($handle)) - 1, 0);
rewind($handle);
$lines = 0;
$rocks = 0;
while (($line = fgets($handle)) != false) {
	$lines++;
	for ($i = 0; $i < strlen($line); $i++) {
		switch ($line[$i]) {
			case 'O':
				$sum += $columns[$i]++;
				$rocks++;
				break;
			case '#':
				$columns[$i] = $lines;
				break;
		}
	}
}
fclose($handle);

$sum = $rocks * $lines - $sum;

echo $sum . PHP_EOL;
